flink-core/src/main/java/org/apache/flink/configuration/TaskManagerOptions.java

```diff
@@ -571,5 +573,9 @@ private TaskManagerOptions() {}
 	public static class ManagedMemoryConsumerNames {
```

Comment:
It may be out of the scope of this task.
But I'm wondering whether we move this class into runtime module and just use string in the config option?
`TaskManagerOptions` is a public interface. Users may also treat `ManagedMemoryConsumerNames` as public interface which is actually unwanted.

Reply:
I think you are right. We should try to avoid adding unnecessary logics/functionalities to `flink-core`. I'll move the class out.